test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,463 @@
+    },
+    appID = c.getHMIAppId(),
+    timeout = 10000
+  })
+  :Do(function(_, data)
+    pResponseFunc(data, pParams)
+    c.getHMIConnection():SendNotification("UI.OnSystemContext",
+      { appID = c.getHMIAppId(), systemContext = "MAIN" },
+      { appID = c.getHMIAppId(), systemContext = "HMI_OBSCURED" })
+  end)
+  c.getMobileSession():ExpectResponse(cid, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+function c.SetInteriorVehicleData( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
+  if not pRespParams then pRespParams = { success = true, resultCode = "SUCCESS" } end
+  local cid = c.getMobileSession():SendRPC(commonRC.getAppEventName("SetInteriorVehicleData"),
+    commonRC.getAppRequestParams("SetInteriorVehicleData", "CLIMATE" ))
+  EXPECT_HMICALL(commonRC.getHMIEventName("SetInteriorVehicleData"),
+    commonRC.getHMIRequestParams("SetInteriorVehicleData", "CLIMATE"))
+  :Do(function(_, data)
+    pParams.respParams =  commonRC.getHMIResponseParams("SetInteriorVehicleData", "CLIMATE")
+    pResponseFunc(data, pParams)
+  end)
+  c.getMobileSession():ExpectResponse(cid, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+function c.rpcAllowedWithConsent( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
+  if not pRespParams then pRespParams = { success = true, resultCode = "SUCCESS" } end
+  local cid = c.getMobileSession(2):SendRPC(commonRC.getAppEventName("SetInteriorVehicleData"),
+    commonRC.getAppRequestParams("SetInteriorVehicleData", "CLIMATE"))
+
+  local consentRPC = "GetInteriorVehicleDataConsent"
+  EXPECT_HMICALL(commonRC.getHMIEventName(consentRPC), commonRC.getHMIRequestParams(consentRPC, "CLIMATE", 2))
+  :Do(function(_, data)
+    pParams.respParams = commonRC.getHMIResponseParams(consentRPC, true)
+    pResponseFunc(data, pParams)
+  end)
+  EXPECT_HMICALL(commonRC.getHMIEventName("SetInteriorVehicleData"),
+  commonRC.getHMIRequestParams("SetInteriorVehicleData", "CLIMATE", 2))
+  :Do(function(_, data)
+    c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+      commonRC.getHMIResponseParams("SetInteriorVehicleData", "CLIMATE"))
+  end)
```

Comment:
Please add timeout for SetInteriorVehicleData, becase in case GetInteriorVehicleDataConsent response will be sent with delay expectation of the SetInteriorVehicleData is expired, like 
```
EXPECT_HMICALL(commonRC.getHMIEventName("SetInteriorVehicleData"),
  commonRC.getHMIRequestParams("SetInteriorVehicleData", "CLIMATE", 2))
  :Do(function(_, data)
    c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
      commonRC.getHMIResponseParams("SetInteriorVehicleData", "CLIMATE"))
  end)
  :Timeout(pRespExpect)
```